pkg/controller/volume/attach_detach_controller.go

```diff
@@ -19,63 +19,109 @@ limitations under the License.
 package volume
 
 import (
+	"fmt"
+	"strings"
 	"time"
 
 	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/cloudprovider"
 	"k8s.io/kubernetes/pkg/controller/framework"
 	"k8s.io/kubernetes/pkg/controller/framework/informers"
+	"k8s.io/kubernetes/pkg/controller/volume/attacherdetacher"
+	"k8s.io/kubernetes/pkg/controller/volume/cache"
+	"k8s.io/kubernetes/pkg/controller/volume/reconciler"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/io"
+	"k8s.io/kubernetes/pkg/util/mount"
 	"k8s.io/kubernetes/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/volume"
 )
 
-// AttachDetachController defines the operations supported by this controller.
-type AttachDetachController interface {
-	Run(stopCh <-chan struct{})
-}
+const (
+	// ControllerManagedAnnotation is the key of the annotation on Node objects
+	// that indicates attach/detach operations for the node should be managed
+	// by the attach/detach controller
+	ControllerManagedAnnotation string = "volumes.kubernetes.io/controller-managed-attach-detach"
 
-type attachDetachController struct {
-	// internalPodInformer is the shared pod informer used to fetch and store
-	// pod objects from the API server. It is shared with other controllers and
-	// therefore the pod objects in its store should be treated as immutable.
-	internalPodInformer framework.SharedInformer
+	// SafeToDetachAnnotation is the annotation added to the Node object by
+	// kubelet in the format "volumes.kubernetes.io/safetodetach/{volumename}"
+	// to indicate the volume has been unmounted and is safe to detach.
+	SafeToDetachAnnotation string = "volumes.kubernetes.io/safetodetach/"
 
-	// selfCreatedPodInformer is true if the internalPodInformer was created
-	// during initialization, not passed in.
-	selfCreatedPodInformer bool
+	// loopPeriod is the ammount of time the reconciler loop waits  between
+	// successive executions
+	reconcilerLoopPeriod time.Duration = 100 * time.Millisecond
```

Comment:
The amount of time the reconciler waits between successive executions. Each execution looks at "actual state of world" and "desired state of world" and triggers actions required to reconcile.